python/ray/tune/search/variant_generator.py

```diff
@@ -59,11 +59,38 @@ def generate_variants(
 
 
 @PublicAPI(stability="beta")
-def grid_search(values: Iterable) -> Dict[str, List]:
-    """Convenience method for specifying grid search over a value.
+def grid_search(values: Iterable) -> Dict[str, Iterable]:
+    """Specify a grid of values to search over.
+
+    Values specified in a grid search are guaranteed to be sampled.
+
+    If multiple grid search variables are defined, they are combined with the
+    combinatorial product. This means every possible combination of values will
+    be sampled.
+
+    Example:
+
+        >>> from ray import tune
+        >>> param_space={
+        ...   "x": tune.grid_search([10, 20]),
+        ...   "y": tune.grid_search["a", "b", "c"])
+        ... }
+
+        This will create a grid of 6 samples:
+        ``{"x": 10, "y": "a"}, {"x": 10, "y": "b"}, `` etc.
+
+    When specifying ``num_samples`` in the
+    :class:`TuneConfig <ray.tune.tune_config.TuneConfig>` in addition to
+    grid search parameters, the number of samples will be drawn for each grid
+    parameter.
```

Comment:
Nit: I got kinda confused reading this sentence. Like, I only understood what it meant after reading the example below.

Reply:
Reworded a bit, hope this makes it clear!

Reply:
It's much clearer, thanks!